Fixed toggle shortcut and added in TkInterface too

umutdz opened the pull request:
…ace as well

e3rd commented on mininterface/tk_interface/tk_window.py:
```diff
@@ -31,6 +30,8 @@ def __init__(self, interface: "TkInterface"):
         self.interface = interface
         self.title(interface.title)
         self.bind('<Escape>', lambda _: self._ok(Cancelled))
+        self.bind('<Control-h>', self._show_help)  # Help with Ctrl+H
```
now, a help shows the hint for secret even if no secret is on the page. I prefer not having such help :)

e3rd commented on mininterface/tk_interface/tk_window.py:
```diff
@@ -31,6 +30,8 @@ def __init__(self, interface: "TkInterface"):
         self.title(interface.title)
         self.bind('<Escape>', lambda _: self._ok(Cancelled))
+        self.bind('<Control-h>', self._show_help)  # Help with Ctrl+H
+        self.bind('<Control-t>', self._toggle_secrets)
```
this imposes a global shortcut. Take look at date_entry – there, the shortcut works only if we are in the field.

That way, it toogles all the secrets in the form, not a single one.

```python
@dataclass
class Credentials:
    password: Annotated[str, SecretTag()]
    api_key: Annotated[str, SecretTag(show_toggle=True)]


m = run(interface="gui")
out = m.form({
    "Password": SecretTag(""),
    "API Key": SecretTag("", show_toggle=True)  # with eye icon
})
print("51: out =", out)
```

e3rd commented on mininterface/tk_interface/utils.py:
```diff
@@ -150,7 +150,10 @@ def _fetch(variable):
+                    # Create wrapper and store it in the widget list
+                    wrapper = SecretEntryWrapper(master, tag, variable, grid_info)
+                    widget = wrapper.entry
+                    subwidgets.append(wrapper)
```
I think it was better before.
Subwidgets seems to be used only for real entries, not wrappers. Or what this should do? :)

umutdz commented:
Thanks for the comments. It's all fixed!

e3rd reviewed (changes requested):
You're insane! I appreciate the help window now! Instead of `_secret_wrapper` storing self, let it be ex. `_shortcut_hint` having the direct text.

That way, you can add the same `_shortcut_hint` for DateEntry too, will you?

umutdz commented:
Make sense, I'll do it

umutdz commented:
When collecting shortcuts for the help window, the code traverses both form fields and their associated widgets. Having the hints available at both levels ensures they're discoverable during this traversal.
I had to change again to use subwidgets.

e3rd commented:
* Why do you use subwidgets, what for? :)
* Why storing the shortcut both to the field_form and to the widget?
* When having multiple SecretTags (as in my last example), there are two same hints in the window.
* You may use `flatten(widgets_to_dict(form))` for recursively iterate all widgets, without bothering to recurse.

I think the best would be to have `TkWindow.shortcuts = set()`  attribute that you store all the shortcuts to. That way, you don't have to use private non-__init__ defined attributes (like _shortcut) and you don't bother when the shortcut appears twice! :)

umutdz commented:
Absolutely, I was making it more complicated than necessary :)

I've simplified it by using a central shortcuts set in TkWindow, which automatically handles duplicates. No more storing shortcuts in multiple places or complex widget traversal. Also removed the unused subwidgets stuff since we don't really need it for the wrapper.

e3rd commented:
That's great, thanks a lot for your contribution! :) 

Is there any other part of the code that you're willing to use and thus want to improve? Ex. path dialog and datetime handling do not work in textual at all.

umutdz commented:
Thanks for merging; it's a good idea. I haven't experienced this problem. Can you open an issue and explain which part you want to be improved

e3rd commented:
Here I put an idea what might be done for the case you are interested https://github.com/CZ-NIC/mininterface/issues/17
